### PR TITLE
Listener: Connections accepted per socket event histogram.

### PR DESCRIPTION
**Pull request by KBaichoo**
Commit Message: Follow up adding a histogram of connections accepted per socket event.
Additional Description: https://github.com/envoyproxy/envoy/pull/27773#discussion_r1222002852
Risk Level: low / med
Testing: unit, integration
Docs Changes: included
Release Notes: included
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by KBaichoo** [comment]
/assign @ggreenway 

Since you have the underlying context

File: changelogs/current.yaml [hunk 1]
```diff
@@ -225,7 +225,9 @@ new_features:
     that sets the maximum number of new connections to be accepted per socket
     event on a listener. If there are more connections to be accepted beyond
     the maximum, the remaining connections would be processed in later
-    dispatcher loop iterations.
+    dispatcher loop iterations. Added listener histogram
+    ``connections_accepted_per_socket_event`` to allow users to empperically
```
Comment (soulxu) [inline changelogs/current.yaml:229]: s/empperically/empirically ?, A new English word for me, at least my dict doesn't find this word :)

**Review by soulxu — APPROVED** [review]
LGTM, except it seems a typo here.

**Comment by KBaichoo** [comment]
Thanks for the reviews @ggreenway and @soulxu!

**Pull request merged**